src/hooks/govwallet/useGovWalletBalance/useGovWalletBalance.test.tsx

```diff
@@ -0,0 +1,158 @@
+import { renderHook } from "@testing-library/react-hooks";
+import React, { FunctionComponent } from "react";
+import { getGovWalletBalance } from "../../../services/govwallet/balance";
+import {
+  defaultCampaignConfig,
+  defaultFeatures,
+  defaultProducts,
+} from "../../../test/helpers/defaults";
+import { CampaignConfig } from "../../../types";
+import { useGovWalletBalance } from "./useGovWalletBalance";
+import { ProductContextProvider } from "../../../context/products";
+import { CampaignConfigContextProvider } from "../../../context/campaignConfig";
+import { NetworkError } from "../../../services/helpers";
+
+jest.mock("../../../services/govwallet/balance");
+const mockGetGovWalletBalance = getGovWalletBalance as jest.Mock;
+
+const key = "KEY";
+const endpoint = "https://myendpoint.com";
+
+const testCampaignConfig: CampaignConfig = {
+  ...defaultCampaignConfig,
+  features: {
+    ...defaultFeatures,
+    checkGovWalletBalance: true,
+  },
+};
+
+const wrapper: FunctionComponent = ({ children }) => (
+  <ProductContextProvider products={defaultProducts}>
+    <CampaignConfigContextProvider campaignConfig={testCampaignConfig}>
+      {children}
+    </CampaignConfigContextProvider>
+  </ProductContextProvider>
+);
+
+const wrapperWithToggleOff: FunctionComponent = ({ children }) => (
+  <ProductContextProvider products={defaultProducts}>
+    <CampaignConfigContextProvider campaignConfig={defaultCampaignConfig}>
+      {children}
+    </CampaignConfigContextProvider>
+  </ProductContextProvider>
+);
+
+let ids: string[];
+
+describe("checkGovWalletBalance toggle", () => {
+  beforeAll(() => {
+    ids = ["S0000000J"];
+
+    mockGetGovWalletBalance.mockResolvedValue({
+      accountDetails: [{ balance: 10000 }],
+    });
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should get balance if toggle is true", async () => {
+    expect.assertions(3);
+
+    const { result, waitForNextUpdate } = renderHook(
+      () => useGovWalletBalance(ids, key, endpoint),
+      { wrapper }
+    );
+
+    expect(result.current.govWalletBalanceState).toStrictEqual("DEFAULT");
+    await waitForNextUpdate();
+    expect(result.current.govWalletBalanceState).toStrictEqual("SUFFICIENT");
+    expect(mockGetGovWalletBalance).toHaveBeenCalledTimes(1);
+  });
+
+  it("should NOT get balance if toggle is false", async () => {
+    expect.assertions(2);
+
+    const { result } = renderHook(
+      () => useGovWalletBalance(ids, key, endpoint),
+      { wrapper: wrapperWithToggleOff }
+    );
+
+    expect(result.current.govWalletBalanceState).toStrictEqual("DEFAULT");
+    expect(mockGetGovWalletBalance).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe("govWalletBalanceState states", () => {
+  beforeAll(() => {
+    ids = ["S0000000J", "S0000001I"];
```

Comment:
Question: when are there multiple ids?

Reply:
There can be multiple IDs when the campaign allows for checkout for multiple users at the same time. Since this is not the case, we can simplify the tests!

Reply:
Currently, the hook loops through the `ids` and retrieves the balance for each of them, do you think we should simplify it and only retrieve the balance for the first id or keep it as is?

I vote to keep it as is since it is already implemented 🙈

Reply:
Ok to leave as is.